File: Python/quicksort.py
arr = [10, 7, 8, 9, 1, 5] 


def swap(a,b):
	temp=arr[a]
	arr[a]=arr[b]
	arr[b]=temp

def partition(arr,low,high):
	pivot=arr[high]
	i=(low-1)
	for j in range(low,high):
		if(arr[j]<=pivot):
			i=i+1
			swap(i,j)
	i=i+1
	swap(i,high)
	return i

def quicksort(arr,low,high):
	if(low<high):
		pi=partition(arr,low,high)
		quicksort(arr,low,pi-1)
		quicksort(arr,pi+1,high)


quicksort(arr,0,len(arr)-1)

print(arr)

File: Javascript/quicksort.js
let arr = [10, 80, 30, 90, 70, 50, 40];

swap=(a,b)=>{
	let temp=arr[a];
	arr[a]=arr[b];
	arr[b]=temp;
}

partition=(low,high)=>{
	let pivot = arr[high];
	let i = low-1;

	for(let j=low;j<high;j++){

		if(arr[j]<pivot){
			i++;
			swap(i,j);
		}

	}
	i++;
	swap(i,high);
	return i;
}


quicksort=(low,high)=>{
	if(low<high){
		let pi=partition(low,high);
		quicksort(low,pi-1);
		quicksort(pi+1,high);
	}
}

quicksort(0,arr.length-1);
console.log("Sorted Array: ",arr);